### PR TITLE
feat: reth-config crate

### LINKED ISSUE #2809: Add reth-config crate with all config (reth.toml) types
### Describe the feature

Config currently resides in 

https://github.com/paradigmxyz/reth/blob/5039122c356f20d36e3d5979f0da97a2fc40ac75/crates/staged-sync/src/config.rs#L12-L23

we want a standalone crate that provides all config types.

## TODO
- [ ] add reth-config crate
- [ ] move Config struct to reth-config

ref #2514

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by bowenyou**
Closes https://github.com/paradigmxyz/reth/issues/2809.

### CODE REVIEW

File: crates/staged-sync/src/config.rs
Comment (onbjerg) [inline crates/staged-sync/src/config.rs:1]: Everything in this file should be removed, it should be removed from `reth-staged-sync` in its entirety

**Review by onbjerg — CHANGES_REQUESTED** [review]

File: crates/staged-sync/src/config.rs
Comment (bowenyou) [reply crates/staged-sync/src/config.rs:1]: okay, moved all of it over to `reth-config` now

**Review by onbjerg — APPROVED** [review]
lgtm, @mattsse does this resolve your original issue? i guess other config changes are separate prs?

**Review by mattsse — APPROVED** [review]
lgtm

> i guess other config changes are separate prs?

yes, I believe so

**Pull request merged**